### PR TITLE
merge mainline changes back into dev

### PR DESCRIPTION
**Pull request by psychocoderHPC**
After the last release, we have not merged the master back into the dev branch. This PR is doing it to avoid merge conflicts when we release next.

```
git merge -X ours --no-ff mainline/master
```

### CODE REVIEW

File: share/picongpu/dockerfiles/README.rst [hunk 1]
```diff
@@ -56,7 +56,7 @@ You can also push the result to dockerhub and singularity-hub (you need an accou
     cd ubuntu-1604
 
     # docker image
-    docker build -t ax3l/picongpu:0.5.0 .
```
Comment (psychocoderHPC) [inline share/picongpu/dockerfiles/README.rst:59]: This change is wrong and introduced in https://github.com/ComputationalRadiationPhysics/picongpu/commit/0cc2a2291813e1012b4a10e8f029ce5b80cba6ca#diff-821c08a5fdb3c6cb5a0ea2a56bd807feeb65562430d68ae637a6b2d92f6db03e

I will fix it after this PR is merged!

Comment (PrometheusPi) [reply share/picongpu/dockerfiles/README.rst:59]: Okay

File: include/picongpu/version.hpp [hunk 2]
```diff
@@ -22,4 +22,4 @@
 #define PICONGPU_VERSION_MAJOR 0
 #define PICONGPU_VERSION_MINOR 5
 #define PICONGPU_VERSION_PATCH 0
-#define PICONGPU_VERSION_LABEL "dev"
+#define PICONGPU_VERSION_LABEL ""
```
Comment (PrometheusPi) [inline include/picongpu/version.hpp:25]: should this not be "dev"?

**Review by PrometheusPi — DISMISSED** [review]
Just a question

File: include/picongpu/version.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/version.hpp:25]: right this should not be part of this PR

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**